### PR TITLE
fix(extractor: odnoklassniki): for download URL use the same protocol as in original URL

### PR DESCRIPTION
**Pull request by actionless**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Some videos are not downloading well via http. While investigating the problem i've noticed what when i change the hardcoded protocol from http to https those videos downloading well.

### CODE REVIEW

**Comment by dstftw** [comment]
1. Provide concrete example URLs.
2. Add a test.

**Comment by actionless** [comment]
@dstftw done

File: youtube_dl/extractor/odnoklassniki.py [hunk 1]
```diff
@@ -121,6 +121,10 @@ class OdnoklassnikiIE(InfoExtractor):
         # Paid video
         'url': 'https://ok.ru/video/954886983203',
         'only_matching': True,
+    }, {
+        # Downloads only via https
+        'url': 'https://ok.ru/video/1704000096969',
```
Comment (dstftw) [inline youtube_dl/extractor/odnoklassniki.py]: >Video has been blocked due to author's rights infingement

Comment (actionless) [reply youtube_dl/extractor/odnoklassniki.py]: updated the link

Comment (dstftw) [reply youtube_dl/extractor/odnoklassniki.py]: https://ok.ru/video/1705664645833 downloads fine without these changes.

Comment (actionless) [reply youtube_dl/extractor/odnoklassniki.py]: for me without this change it returns:
```
[Odnoklassniki] 1705664645833: Downloading webpage
ERROR: Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

while for some other URLs i've tried it works

Comment (actionless) [reply youtube_dl/extractor/odnoklassniki.py]: after the latest update (2020.03.01) the issue is still reproducible for me

**Comment by actionless** [comment]
for some reason the problem is reproducible only from inside centralwoonen's network (who go outside through ziggo), but not even reproducible from other networks served by ziggo provider

**Pull request closed without merging**